### DISCUSSION TITLE
Is there a better way to debug server related errors in MCP Inspector

### DISCUSSION THREAD

**Discussion by tech4life87** [modelcontextprotocol/.github#38] [Q&A]
I am currently working on building an MCP server. while debugging using inspector the only error is a Request timeout. I can tell from MCP debug logs that my API call is working. but no meaningful logs I can use to debug further why MCP inspector is timing out. I have the loglevel set at DEBUG

![image](https://github.com/user-attachments/assets/706e29f6-fadd-4faf-92d0-ce69bedfc5ca)

**Comment by dsp-ant** [comment]
The most meaningful way is to esnure that you `@server.list_resource()` handler returns resources by logging them to stderr. I would suggest to open an issue in the inspector repository that we need to improve this scenario. I will go ahead and open a ticket in the python-sdk to improve our logging inside the SDK.
